cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -879,7 +888,44 @@ readSourcePackage verbosity (ProjectPackageLocalDirectory dir cabalFile) = do
       packageSource        = LocalUnpackedPackage (root </> dir),
       packageDescrOverride = Nothing
     }
-readSourcePackage _verbosity _ =
+
+readSourcePackage verbosity distDirLayout (ProjectPackageRemoteRepo repo) = do
+    let
+      scmSrcDir  = "scm-" ++ showHashValue (hashSourceRepo repo)
```

Comment:
I'm not convinced that this is a good idea. I think we ought to be able to pick generally non-clashing names here without having to use huge hashes.

So what's the issue? People said they wanted to be able to have the same repo but with two different checkouts (and presumably two different subdirs otherwise they've selecting two versions of the same package which isn't supported with local dirs either).

So I can see that picking non-clashing names is an issue, but on the other hand we don't want to have to do a fresh checkout when someone updates the project file to point to a different tag. In those cases we should simply be able to fetch & reset. Note that this may imply more functionality in the branchers than we have right now, since currently we can only get a fresh checkout I think right?